### ISSUE TITLE
Error messages for incompatible `into` types are rather inconsistent

### ISSUE THREAD

**Issue by webbedspace** [nushell/nushell#6757]
### Describe the bug

In most of the error messages in the following example, the wording is inconsistent, and moreover, the arrow placement isn't entirely consistent either.

### How to reproduce

```
〉{|A| ls} | into binary
Error: nu::shell::unsupported_input (link)

  × Unsupported input
   ╭─[entry #11:1:1]
 1 │ {|A| ls} | into binary
   ·            ─────┬─────
   ·                 ╰── 'into binary' for unsupported type
   ╰────

〉{|A| ls} | into db
Error: nu::shell::cant_convert (link)

  × Can't convert to database.
   ╭─[entry #12:1:1]
 1 │ {|A| ls} | into db
   · ────┬───
   ·     ╰── can't convert block to database
   ╰────

〉{|A| ls} | into int
Error: nu::shell::unsupported_input (link)

  × Unsupported input
   ╭─[entry #13:1:1]
 1 │ {|A| ls} | into int
   ·            ────┬───
   ·                ╰── 'into int' for unsupported type 'block'
   ╰────

〉{|A| ls} | into string
Error: nu::shell::cant_convert (link)

  × Can't convert to string.
   ╭─[entry #14:1:1]
 1 │ {|A| ls} | into string
   ·            ─────┬─────
   ·                 ╰── can't convert block to string
   ╰────

〉
```

### Expected behavior

Let's see… `'into int' for unsupported type 'block'` is somewhat awkward wording, so I'd prefer the `can't convert block to string` variant of this message. Also, I'd like the arrow to point at the conversion operation, not the "value originated here" operation.

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.69.1                                          |
| branch             |                                                 |
| commit_hash        | 8a5abc7afc28a36908a32f1852127e2a5170b461        |
| build_os           | windows-x86_64                                  |
| build_target       | x86_64-pc-windows-msvc                          |
| rust_version       | rustc 1.64.0 (a55dd71d5 2022-09-19)             |
| rust_channel       | stable-x86_64-pc-windows-msvc                   |
| cargo_version      | cargo 1.64.0 (387270bc7 2022-09-16)             |
| pkg_version        | 0.69.1                                          |
| build_time         | 2022-09-28 02:58:08 +00:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  |                                                 |

**Comment by fdncred** [comment]
good catch. consistency in these spans would be nice to have. you should join [discord](https://discord.gg/NtAbbGn) to discuss some of these issues.

**Comment by webbedspace** [comment]
I don't know any Rust so I don't know how much help I could be with PRs.

**Comment by 132ikl** [comment]
These are all consistent as of 0.107.0, stating "command doesn't support closure input"
